src/main/scala/net/snowflake/spark/snowflake/Parameters.scala

```diff
@@ -196,11 +202,29 @@ object Parameters {
         "A snowflake user must be provided with '" + PARAM_SF_USER + "' parameter, e.g. 'user1'"
       )
     }
+    val authenticatorVal = userParameters.getOrElse(PARAM_AUTHENTICATOR, "")
     if ((!userParameters.contains(PARAM_SF_PASSWORD)) &&
-        (!userParameters.contains(PARAM_PEM_PRIVATE_KEY))) {
+        (!userParameters.contains(PARAM_PEM_PRIVATE_KEY)) &&
+        // 2020-02-13 - if OAuth token not provided
+        (!authenticatorVal.equals("oauth"))) {
```

Comment:
Please explain why to compare it with "oauth"?

Reply:
I cleaned up the check to use the token.  Basically one of 3 things must be provided - token, password, or key